java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -172,14 +175,12 @@ protected void saveResults(Resource testFile, UnitTestClassReport report) {
     }
   }
 
-  protected Resource getUnitTestResource(String classKey) {
+  protected InputFile getUnitTestResource(String classKey) {
     return javaResourceLocator.findResourceByClassName(classKey);
   }
 
-  private static void saveMeasure(SensorContext context, Resource resource, Metric metric, double value) {
-    if (!Double.isNaN(value)) {
-      context.saveMeasure(resource, metric, value);
-    }
+  private static <T extends Serializable> void saveMeasure(SensorContext context, InputFile inputFile, Metric<T> metric, T value) {
+    context.<T>newMeasure().forMetric(metric).on(inputFile).withValue(value).save();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)